pkg/crc/validation/validation.go

```diff
@@ -97,6 +97,10 @@ func ValidateBundle(bundlePath string, preset crcpreset.Preset) error {
 		}
 		return ValidateBundlePath(bundlePath, preset)
 	}
+	if err = BundleMismatchWithPreset(preset, bundleMetadata); err != nil {
```

Comment:
while testing its not showing the expected `FATAL` log message when doing `crc setup` with a bundle of different preset as the previous line has `return ValidateBundlePath(bundlePath, preset)`

```
% crc config view
- bundle                                : docker://quay.io/crcont/openshift-bundle:4.16.7
- consent-telemetry                     : no
- preset                                : microshift

% crc setup --bundle ~/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle
WARN Using crc_vfkit_4.16.7_arm64.crcbundle bundle, but crc_microshift_vfkit_4.16.7_arm64.crcbundle is expected for this release
INFO Using bundle path /Users/anath/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle
INFO Checking if running macOS version >= 13.x
INFO Checking if running as non-root
INFO Checking if crc-admin-helper executable is cached
INFO Checking if running on a supported CPU architecture
INFO Checking if crc executable symlink exists
INFO Checking minimum RAM requirements
INFO Check if Podman binary exists in: /Users/anath/.crc/bin/oc
INFO Checking if running emulated on Apple silicon
INFO Checking if vfkit is installed
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if /Users/anath/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle exists
INFO Getting bundle for the CRC executable
INFO Downloading bundle: /Users/anath/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle...
INFO Uncompressing /Users/anath/.crc/cache/crc_vfkit_4.16.7_arm64.crcbundle
crc.img:  4.71 GiB / 31.00 GiB [----------------------->___________________________________________________________________________________________________________________crc.img:  5.44 GiB / 31.00 GiB [---------------------->____________________________________________________________________________________________________________] 17.54%

```

Reply:
Good catch! I didn't think there would be another code path that is executed when the bundle is uncompressed.
However, I think it's misleading to end `setup` with "your system is correctly set up for crc" and then on the next run fail fatally.
Should we add another bundle check after the decompression is done?

Reply:
when the bundle doesn't exist yet it'd be better to not even download the bundle for the wrong preset, otherwise it feels a bit weird to have it download and uncompress the bundle and later end with a fatal error

so maybe we should have the  `BundleMismatchWithPreset` check in `ValidateBundlePath` as well

Reply:
Correct me if I'm wrong, but before uncompressing it, there is no way to know what preset the bundle is for, no? Therefore the check cannot be done in `ValidateBundlePath`, since this function is called when `bundle.Get(bundleName)` returns `nil`.

Reply:
The metadata file is stored at the beginning of the bundle so in theory you can uncompress it after fetching a few kilobytes of data, but I don't think we have what's needed in crc to easily do this.

Another option is to do some filename based check (`crc_microshift_libvirt_4.16.0.crcbundle`), but that's quite fragile.

Reply:
@redbeam maybe doing filename based check is the easier/quicker solution for this now, or did you want to look at extracting just the metadata file from the bundle archive?

Reply:
@anjannath yes, that is the path I went on in the end. I pushed the changes now.

Reply:
thanks, tested and working as expected!